tests/e2e/test_pvc_creation_deletion_scale.py

```diff
@@ -0,0 +1,224 @@
+"""
+Test to measure pvc scale creation & deletion time. Total pvc count would be 1500
+"""
+import logging
+import csv
+import random
+import pytest
+import threading
+
+from tests import helpers
+from ocs_ci.framework.testlib import scale, E2ETest, polarion_id, ignore_leftovers
+from ocs_ci.ocs import constants
+
+log = logging.getLogger(__name__)
+
+
+@scale
+@ignore_leftovers
+class TestPVCCreationScale(E2ETest):
+    """
+    Base class for PVC scale creation
+    """
+    @pytest.fixture()
+    def namespace(self, project_factory):
+        """
+        Create a new project
+        """
+        proj_obj = project_factory()
+        self.namespace = proj_obj.namespace
+
+    @pytest.mark.parametrize(
+        argnames=["access_mode", "interface"],
+        argvalues=[
+            pytest.param(
+                *[constants.ACCESS_MODE_RWO, constants.CEPHBLOCKPOOL],
+                marks=pytest.mark.polarion_id("OCS-1225")
+            ),
+            pytest.param(
+                *[constants.ACCESS_MODE_RWX, constants.CEPHBLOCKPOOL],
+                marks=pytest.mark.polarion_id("OCS-2010")
+            ),
+            pytest.param(
+                *[constants.ACCESS_MODE_RWO, constants.CEPHFS_INTERFACE],
+                marks=pytest.mark.polarion_id("OCS-2011")
+            ),
+            pytest.param(
+                *[constants.ACCESS_MODE_RWX, constants.CEPHFS_INTERFACE],
+                marks=pytest.mark.polarion_id("OCS-2008")
+            ),
+        ]
+    )
+    @pytest.mark.usefixtures(namespace.__name__)
+    def test_1500_pvc_creation_scale(self, namespace, access_mode, interface):
+        """
+        Measuring PVC creation time while scaling PVC
+        Measure PVC deletion time after creation test
+        """
+        number_of_pvc = 1500
+        log.info(f"Start creating {access_mode}-{interface} {number_of_pvc} PVC")
+
+        if interface == constants.CEPHBLOCKPOOL:
+            self.sc_obj = constants.DEFAULT_STORAGECLASS_RBD
+        elif interface == constants.CEPHFS_INTERFACE:
+            self.sc_obj = constants.DEFAULT_STORAGECLASS_CEPHFS
+
+        # Create PVC
+        pvc_objs = helpers.create_multiple_pvcs(
+            sc_name=self.sc_obj,
+            namespace=self.namespace,
+            number_of_pvc=number_of_pvc,
+            size=f"{random.randrange(5, 105, 5)}Gi",
+            access_mode=access_mode
+        )
+
+        # Check for PVC status using threads
+        threads = list()
+        for obj in pvc_objs:
```

Comment:
Tired ThreadPoolExecutor,, comparatively ThreadPoolExecutor took lot of time than Threads. thread evaluation will be faster, i.e. Since it's in for loop before reaching ~100th thread,, first thread will be completed. 
Execution of this testcase in magna machine didn't throw any error due to threads. in local laptop tried execution with 500, but again same logic will apply.
There will be a problem if PVC not in bound state.

Reply:
IMO there is no need for threads at all.
1) probably after the creation of all 1500 PVCs, the first few hundreds already bounded.
2) when you'll reach to a PVC which you are waiting for, the rest will get closer to be bounded 

So in overall, you will wait for just few PVC instead of having all those threads.

Reply:
Yes, that's right. Sequentially checking bound state for each PVC is going to take time, performing this operation in threads for parallelism.